src/transformers/configuration_blenderbot.py

```diff
@@ -0,0 +1,157 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# LICENSE file in the root directory of this source tree.
+from .configuration_utils import PretrainedConfig
+
+
+BLENDERBOT_PRETRAINED_CONFIG_ARCHIVE_MAP = {
+    "facebook/blenderbot-3B": "https://cdn.huggingface.co/facebook/blenderbot-3B/config.json",
+    "facebook/blenderbot-90M": "https://cdn.huggingface.co/facebook/blenderbot-/config.json",
+}
+
+
+class BlenderbotConfig(PretrainedConfig):
+    """
+        This is the configuration class to store the configuration of a :class:`~transformers.BlenderbotModel`.
+        It is used to instantiate an Blenderbot model according to the specified arguments, defining the model
+        architecture. Instantiating a configuration with the defaults will yield a similar configuration to that of
+        the `blenderbot <https://huggingface.co/blenderbot>`__ architecture.
+
+        Configuration objects inherit from  :class:`~transformers.PretrainedConfig` and can be used
+        to control the model outputs. Read the documentation from  :class:`~transformers.PretrainedConfig`
+        for more information.
+
+        Args:
+            d_model: (:obj:`int`, default to 2560), dimension of the embeddings vector
+            encoder_layers: (:obj:`int`, default to 2), number of layers in the encoder
+            encoder_ffn_size: (:obj:`int`, default to 10240), size of hidden layers in the FFN in the encoder
+            decoder_layers: (:obj:`int`, default to 24), number of layers in the decoder
+            decoder_ffn_size: (:obj:`int`, default to 10240), size of hidden layers in the FFN in the decoder
+            dropout: (:obj:`float`, default to 0.1), embedding dropout
+            activation_dropout: (:obj:`float`, default to 0.0), dropout after activation function
+            encoder_layerdrop: (:obj:`float`, default to 0.0,
+            decoder_layerdrop: (:obj:`float`, default to 0.0),
+            encoder_attention_heads:(:obj:`int`, default to 32),  number of multi heads attention in the encoder
+            decoder_attention_heads:(:obj:`int`, default to 32),  number of multi heads attention in the encoder
+            max_positions_embeddings:(:obj:`int`, default to 128), size of the position embeddings
+            activation: (:obj:`string`, default to 'gelu'), activation function to use
+            attention_dropout: (:obj:`float`, default to 0.0), multi head attention dropout
+            relu_dropout: (:obj:`float`, default to 0.0), relu dropout
+            vocab_size: (:obj:`int`, default to 8008), the size of the vocabulary
+            static_position_embeddings: (:obj:`boolean`, default to False),  if yes or no the positional embeddings will be learn
+            variant: (obj: str, default to "prelayernorm") defines when to apply a layernorm
```

Comment:
this param should be explained nicely. What other options are there? How do they differ from each other?